presentation/src/main/java/com/anytypeio/anytype/presentation/history/VersionHistoryViewModel.kt

```diff
@@ -468,34 +513,111 @@ class VersionHistoryViewModel(
                         .first()
                     val obj =
                         ObjectWrapper.Basic(event.details.details[vmParams.objectId]?.map.orEmpty())
-                    val root = event.blocks.first { it.id == vmParams.objectId }
-                    val blocks = event.blocks.asMap().render(
-                        mode = Mode.Read,
-                        root = root,
-                        focus = Editor.Focus.empty(),
-                        anchor = vmParams.objectId,
-                        indent = INITIAL_INDENT,
-                        details = event.details,
-                        relationLinks = event.relationLinks,
-                        restrictions = event.objectRestrictions,
-                        selection = emptySet()
-                    )
                     val currentState = _previewViewState.value
                     if (currentState !is VersionHistoryPreviewScreen.Hidden) {
-                        _previewViewState.value = VersionHistoryPreviewScreen.Success(
-                            versionId = item.id,
-                            blocks = blocks,
-                            dateFormatted = item.dateFormatted,
-                            timeFormatted = item.timeFormatted,
-                            icon = item.icon,
-                            isSet = obj.layout.isDataView()
+                        parseObject(
+                            payload = payload,
+                            event = event,
+                            item = item,
+                            obj = obj
                         )
                     }
                 }
             }
         )
     }
 
+    private suspend fun parseObject(
+        payload: Payload,
+        event: Event.Command.ShowObject,
```

Comment:
let's add a todo to refact `showVersion` response to include `ObjectView` instead of `Payload`. Here's a similar example:

```
class OpenObject @Inject constructor(
    private val repo: BlockRepository,
    private val settings: UserSettingsRepository,
    dispatchers: AppCoroutineDispatchers
) : ResultInteractor<OpenObject.Params, ObjectView>(dispatchers.io)
```

Reply:
added